### DISCUSSION TITLE
Artillery ignoring settings in phases section when testing socket.io server

### DISCUSSION THREAD

**Discussion by afrancis-caregility** [artilleryio/artillery#1835] [Artillery]
This may not be a bug and perhaps an issue with how I am testing my socket.io server. But it is a simple scenario, and I am not sure what I could be doing wrong in my yaml file.
I am trying to load test my local socket.io server and the test when run only creates one vuser even though I have specified duration = 10 and arrivalRate = 5

Version info: 
```
Artillery version - 2.0.0-30
on a MacBook Pro with M1 chip

```
<img width="679" alt="image" src="https://user-images.githubusercontent.com/103443530/220159733-89a22c08-8eda-40dc-b064-9c7ea2d59d42.png">

Running this command: 
```
DEBUG=socketio artillery run scripts/artillery-load-test.yaml
```

I expected to see this happen:
I expected to see many socket clients created but I always only see one socket connection.

<img width="418" alt="image" src="https://user-images.githubusercontent.com/103443530/220159550-15c7f454-950b-4a82-9ca2-8fe0d4e872ed.png">

Instead, this happened:

only one vuser is created.
<img width="755" alt="image" src="https://user-images.githubusercontent.com/103443530/220159337-68b62bec-1533-4b89-b38d-c5173174d253.png">

Files being used:
```
artillery-load-test.yaml
```

**Comment by hassy** [comment]
hi @afrancis-caregility 👋 The fix is to move `phases` to be directly under `config` rather than under `config.socketio`. That's where Artillery expects to find phase definitions, and if it doesn't it creates just one VU (this is to make testing new scripts easier).

**Reply by afrancis-caregility** [reply]
Thanks that fixed it. :)

**Comment by hassy** [comment]
moving to discussions

**Comment by afrancis-caregility** [comment]
Hi @hassy , The randomNumber and randomString functions that I used in my script only get invoked once and are the same for all the requests. Anyway, to make it random for all the requests? I want to pass random socket.io query parameters for each vuser created rather than just one user.

**Comment by afrancis-caregility** [comment]
Also, multiple arguments for emit. Is it still not supported with the latest version? 
![image](https://user-images.githubusercontent.com/103443530/220374843-47f331de-43ea-4437-ae7e-015566600cd8.png)
